packages/runtime-core/src/rendererTemplateRef.ts

```diff
@@ -128,9 +130,11 @@ export function setRef(
             }
           }
         } else if (_isString) {
-          refs[ref] = value
-          if (canSetSetupRef(ref)) {
-            setupState[ref] = value
+          if (oldRef !== ref) {
```

Comment:
Maybe we can issue a runtime warning when `oldRef === ref`.

Reply:
Added warning at runtime. 
Ensured that recursion does not occur with the Ref or Ref function type, but only with the string.